src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -225,16 +226,20 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 adjustedInvocationTime = ScheduleStatus.Next;
             }
 
+            // Create the Last value with the adjustedInvocationTime; otherwise, the listener will
+            // consider this a schedule change when the host next starts.
             ScheduleStatus = new ScheduleStatus
             {
-                Last = invocationTime,
+                Last = adjustedInvocationTime,
                 Next = _schedule.GetNextOccurrence(adjustedInvocationTime)
             };
 
             if (ScheduleMonitor != null)
             {
                 await ScheduleMonitor.UpdateStatusAsync(_timerName, ScheduleStatus);
             }
+
+            _trace.Verbose($"Function '{_timerName}' updated status: Last='{ScheduleStatus.Last.ToString("o")}', Next='{ScheduleStatus.Next.ToString("o")}'");
```

Comment:
The other log entry you added is only written when schedule monitoring is on. Let's make this consistent. I assume your intent is to log this info only if schedule monitoring is on? If so, move this into the if block directly above.